Userland/Libraries/LibGfx/ImageFormats/JBIG2Loader.cpp

```diff
@@ -573,34 +573,57 @@ struct SegmentData {
     OwnPtr<Gfx::BilevelImage> aux_buffer;
 };
 
-static void composite_bilevel_image(BilevelImage& out, BilevelImage const& bitmap, Gfx::IntPoint position, JBIG2::CombinationOperator operator_)
+template<JBIG2::CombinationOperator Operator>
+static void composite_bilevel_image_impl(BilevelImage& out, BilevelImage const& bitmap, Gfx::IntRect clip_rect, Gfx::IntPoint position)
 {
-    static constexpr auto combine = [](bool dst, bool src, JBIG2::CombinationOperator op) -> bool {
-        switch (op) {
-        case JBIG2::CombinationOperator::Or:
-            return dst || src;
-        case JBIG2::CombinationOperator::And:
-            return dst && src;
-        case JBIG2::CombinationOperator::Xor:
-            return dst ^ src;
-        case JBIG2::CombinationOperator::XNor:
-            return !(dst ^ src);
-        case JBIG2::CombinationOperator::Replace:
-            return src;
+    for (int y = clip_rect.top(); y < clip_rect.bottom(); ++y) {
+        for (int x = clip_rect.left(); x < clip_rect.right(); ++x) {
+            bool src_bit = bitmap.get_bit(x - position.x(), y - position.y());
+            bool dst_bit = out.get_bit(x, y);
+
+            bool result = [&]() -> bool {
+                if (Operator == JBIG2::CombinationOperator::Or)
+                    return dst_bit || src_bit;
+                if (Operator == JBIG2::CombinationOperator::And)
+                    return dst_bit && src_bit;
+                if (Operator == JBIG2::CombinationOperator::Xor)
+                    return dst_bit ^ src_bit;
+                if (Operator == JBIG2::CombinationOperator::XNor)
+                    return !(dst_bit ^ src_bit);
+                if (Operator == JBIG2::CombinationOperator::Replace)
+                    return src_bit;
+                VERIFY_NOT_REACHED();
```

Comment:
this could be use `if constexpr` instead
which can fully replace the iife
then `VERIFY_NOT_REACHED() -> static_assert(false);` in the last else branch I think